addons/l10n_in_edi/models/account_edi_format.py

```diff
@@ -0,0 +1,575 @@
+                    + tax_details_by_code.get("cess_non_advol_amount", 0.00)) * sign,
+                ),
+                "StCesVal": self._l10n_in_round_value((
+                    tax_details_by_code.get("state_cess_amount", 0.00)
+                    + tax_details_by_code.get("state_cess_non_advol_amount", 0.00)) * sign,
+                ),
+                "RndOffAmt": self._l10n_in_round_value(
+                    sum(line.balance for line in invoice.invoice_line_ids if line.is_rounding_line)),
+                "TotInvVal": self._l10n_in_round_value(
+                    (tax_details.get("base_amount") + tax_details.get("tax_amount")) * sign),
+            },
+        }
+        if invoice.company_currency_id != invoice.currency_id:
+            json_payload["ValDtls"].update({
+                "TotInvValFc": self._l10n_in_round_value(
+                    (tax_details.get("base_amount_currency") + tax_details.get("tax_amount_currency")) * sign)
+            })
+        if saler_buyer.get("seller_details") != saler_buyer.get("dispatch_details"):
+            json_payload.update({
+                "DispDtls": self._get_l10n_in_edi_partner_details(saler_buyer.get("dispatch_details"),
+                    set_vat=False, set_phone_and_email=False)
+            })
+        if saler_buyer.get("buyer_details") != saler_buyer.get("ship_to_details"):
+            json_payload.update({
+                "ShipDtls": self._get_l10n_in_edi_partner_details(saler_buyer.get("ship_to_details"))
+            })
+        if is_overseas:
+            json_payload.update({
+                "ExpDtls": {
+                    "ShipBNo": invoice.l10n_in_shipping_bill_number or "",
+                    "ShipBDt": invoice.l10n_in_shipping_bill_date
+                       and invoice.l10n_in_shipping_bill_date.strftime("%d/%m/%Y") or "",
+                    "Port": invoice.l10n_in_shipping_port_code_id.code or "",
+                    "RefClm": tax_details_by_code.get("igst") and "Y" or "N",
+                    "ForCur": invoice.currency_id.name,
+                    "CntCode": saler_buyer.get("buyer_details").country_id.code or "",
+                }
+            })
+        return json_payload
+
+    @api.model
+    def _l10n_in_prepare_edi_tax_details(self, move, in_foreign=False):
+        def l10n_in_grouping_key_generator(tax_values):
+            base_line = tax_values["base_line_id"]
+            tax_line = tax_values["tax_line_id"]
+            line_code = "other"
+            tax_report_line_state_cess = self.env.ref("l10n_in.tax_report_line_state_cess", False)
+            if any(tag in tax_line.tax_tag_ids for tag in self.env.ref("l10n_in.tax_report_line_cess").sudo().tag_ids):
+                if tax_line.tax_line_id.amount_type != "percent":
+                    line_code = "cess_non_advol"
+                else:
+                    line_code = "cess"
+            elif tax_report_line_state_cess and any(
```

Comment:
@jva-odoo No way to put this on one line?

Reply:
no, it's over 120 lines so

Reply:
Please find a way, it is ugly

Reply:
@jco-odoo okay, done